atomicapp/nulecule/config.py

```diff
@@ -0,0 +1,135 @@
+import copy
+import logging
+
+from atomicapp.constants import (GLOBAL_CONF,
+                                 LOGGER_COCKPIT,
+                                 DEFAULT_PROVIDER,
+                                 NAMESPACE_SEPARATOR)
+from collections import defaultdict
+
+cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
+
+
+class Config(object):
+    """
+    Store config data for a Nulecule or Nulecule component.
+    """
+
+    def __init__(self, namespace='', answers=None, cli=None,
+                 data=None, is_nulecule=False):
+        self._namespace = namespace
+        self._is_nulecule = is_nulecule or False
+        self._parent_ns, self._current_ns = self._split_namespace(self._namespace)
+        self._answers = defaultdict(dict)
+        self._answers.update(answers or {})
+        self._cli = cli or {}
+        self._data = data or defaultdict(dict)
+        self._context = None
+        self._provider = None
+
+    @property
+    def provider(self):
+        """
+        Get provider name.
+
+        Returns:
+            Provider name (str)
+        """
+        if self._provider is None:
+            self._provider = self._answers[GLOBAL_CONF].get('provider')
+            if self._provider is None:
+                self._data[GLOBAL_CONF]['provider'] = DEFAULT_PROVIDER
+                self._provider = DEFAULT_PROVIDER
+
+        return self._provider
+
+    @property
+    def providerconfig(self):
+        """
+        Get provider config info taking into account answers and cli data.
+        """
+        pass
+
+    @property
+    def namespace(self):
+        """
+        Get normalized namespace for this instance.
+
+        Returns:
+            Current namespace (str).
+        """
+        return self._namespace or GLOBAL_CONF
```

Comment:
We have a constant named `DEFAULT_NAMESPACE` why not use that, instead of `GLOBAL_CONF`?